internal/experiment/generation/custom.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2021 GramLabs, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generation
+
+import (
+	"fmt"
+	"math"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	redskyappsv1alpha1 "github.com/thestormforge/optimize-controller/api/apps/v1alpha1"
+	redskyv1beta1 "github.com/thestormforge/optimize-controller/api/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type CustomSource struct {
+	Scenario    *redskyappsv1alpha1.Scenario
+	Application *redskyappsv1alpha1.Application
+}
+
+var _ ExperimentSource = &CustomSource{}
+var _ MetricSource = &CustomSource{}
+
+func (s *CustomSource) Update(exp *redskyv1beta1.Experiment) error {
+	if s.Scenario.Custom.PodTemplate != nil {
+		s.Scenario.Custom.PodTemplate.DeepCopyInto(ensureTrialJobPod(exp))
+	}
+
+	if d := s.Scenario.Custom.InitialDelaySeconds; d > 0 {
+		exp.Spec.TrialTemplate.Spec.InitialDelaySeconds = d
+	}
+
+	if rt := s.Scenario.Custom.ApproximateRuntimeSeconds; rt > 0 {
+		exp.Spec.TrialTemplate.Spec.ApproximateRuntime = &metav1.Duration{Duration: time.Duration(rt) * time.Second}
+	}
+
```

Comment:
I feel like we've got similar patterns in the trial setup job codepath. would it make sense to make these utility functions there so we could reuse the same code?

#notforthispr